src/services/sessions.js

```diff
@@ -43,29 +43,86 @@ module.exports = class SessionsHelper {
 	 */
 
 	static async create(bodyData, loggedInUserId, orgId) {
-		bodyData.mentor_id = loggedInUserId
 		try {
-			const mentorDetails = await mentorExtensionQueries.getMentorExtension(loggedInUserId)
+			// If type is passed store it in upper case
+			bodyData.type && (bodyData.type = bodyData.type.toUpperCase())
+			console.log('<<<<<<<<<<<<<<bodyData.type :', bodyData.type, common.PRIVATE)
+			// If session type is private and mentorId is not passed in request body return an error
+			if (
+				bodyData.type &&
+				bodyData.type == common.PRIVATE &&
+				(!bodyData.mentor_id || bodyData.meeting_id == '')
+			) {
+				return common.failureResponse({
+					message: 'MENTORS_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			// session created_by and updated_by added to session creation data
+			bodyData.created_by = loggedInUserId
+			bodyData.updated_by = loggedInUserId
+			let menteeIdsToEnroll = bodyData.mentees ? bodyData.mentees : []
+			// If body data has mentor id use that id to further checks else use logged user_id
+			const mentorIdToCheck = bodyData.mentor_id || loggedInUserId
+			const isSessionCreatedByManager = !!bodyData.mentor_id
+
+			// Confirm passed id is of a mentor
+			const mentorDetails = await mentorExtensionQueries.getMentorExtension(mentorIdToCheck)
 			if (!mentorDetails) {
 				return common.failureResponse({
 					message: 'INVALID_PERMISSION',
 					statusCode: httpStatusCode.bad_request,
 					responseCode: 'CLIENT_ERROR',
 				})
 			}
-
-			const timeSlot = await this.isTimeSlotAvailable(loggedInUserId, bodyData.start_date, bodyData.end_date)
+			console.log(
+				'mentorDetails : : : :: : :: : : : :',
+				mentorDetails,
+				'++++++++',
+				orgId,
+				'{{{{{{{',
+				mentorDetails.visible_to_organizations.includes(orgId),
+				'((((((((',
+				mentorDetails.visibility !== common.ASSOCIATED
+			)
+			// update mentor Id in session creation data
+			if (!bodyData.mentor_id) {
+				bodyData.mentor_id = loggedInUserId
+			}
+			// else if (
```

Comment:
@VISHNUDAS-tunerlabs why this commented out

Reply:
@rakeshSgr there was a bug in this logic. which nevil will be checking . for local testing I have commented it. now reverted